framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1385,19 +1471,33 @@ int pcm_mmap_transfer(struct pcm *pcm, const void *buffer, unsigned int bytes)
 			break;
 		}
 
-		/* copy frames from buffer */
+		/* copy frames to/from buffer */
 		frames = pcm_mmap_transfer_areas(pcm, (void *)buffer, offset, frames);
 		if (frames < 0) {
 			return oops(pcm, ENOMEM, "write error: avail 0x%x\n", avail);
 		}
 
 		offset += frames;
 		count -= frames;
+
+		/* Start the PCM, if required */
+		if (pcm->flags & PCM_OUT && !pcm->running) {
+			if (pcm_start(pcm) < 0) {
+				return oops(pcm, errno, "start error\n");
+			}
+		}
 	}
 
 	return 0;
 }
 
+/** Write data to the codec using mmap audio buffer
+ * @param[in] pcm A PCM handle
+ * @param[in] data Pointer to the buffer containing data to be written
+ * @param[in] count Size of the data in bytes
+ * @returns On success, written number of frames returned. On failure, a negative number returned
```

Comment:
pcm_mmap_transfer either returns 0 or negative values, so the return values may not match to those in documentation.

Reply:
Fixed